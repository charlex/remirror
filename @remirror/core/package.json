{
  "name": "@remirror/core",
  "description": "Provides the core abstractions for managing your remirror editor functionality",
  "license": "MIT",
  "author": "Ifiok Jr. <ifiokotung@gmail.com>",
  "homepage": "https://github.com/ifiokjr/remirror/tree/master/@remirror/core",
  "repository": "https://github.com/ifiokjr/remirror/tree/master/@remirror/core",
  "version": "0.4.2-canary.2",
  "main": "lib/index.js",
  "module": "lib/dist/core.esm.js",
  "browser": {
    "min-document": false
  },
  "files": [
    "internal",
    "lib",
    "src"
  ],
  "scripts": {
    "api": "api-extractor run",
    "api:local": "api-extractor run --local",
    "build": "run-p build:*",
    "build:babel": "babel src --out-dir lib --root-mode=upward --extensions \".ts,.tsx\" -s",
    "build:declaration": "tsc -p ./tsconfig.prod.json --emitDeclarationOnly --declarationMap",
    "dev": "run-p dev:*",
    "dev:babel": "babel src --out-dir lib --root-mode=upward --extensions \".ts,.tsx\" -s --watch",
    "dev:declaration": "tsc -p ./tsconfig.prod.json --emitDeclarationOnly --declarationMap --watch",
    "document": "node ../api-documenter/lib/start.js markdown --input-folder temp --output-folder ../../docs/api/@remirror",
    "lint": "tslint --project tsconfig.lint.json --config ../../tslint.json",
    "typecheck": "tsc -p ./tsconfig.json --noEmit"
  },
  "types": "lib/index.d.ts",
  "dependencies": {
    "@babel/runtime": "^7.6.0",
    "@remirror/core-constants": "0.4.2-canary.2",
    "@remirror/core-helpers": "0.4.2-canary.2",
    "@remirror/core-types": "0.4.2-canary.2",
    "@remirror/core-utils": "0.4.2-canary.2",
    "@remirror/react-portals": "0.4.2-canary.2",
    "@types/prosemirror-inputrules": "^1.0.2",
    "@types/prosemirror-keymap": "^1.0.1",
    "@types/prosemirror-model": "^1.7.2",
    "@types/prosemirror-state": "^1.2.3",
    "prosemirror-inputrules": "^1.0.4",
    "prosemirror-keymap": "^1.0.1",
    "prosemirror-model": "^1.7.2",
    "prosemirror-state": "^1.2.4",
    "prosemirror-suggest": "0.4.2-canary.2"
  },
  "peerDependencies": {
    "prosemirror-view": "^1.11.1",
    "@types/prosemirror-view": "^1.9.0",
    "@emotion/core": "^10",
    "@types/react": "^16.8.0",
    "react": "^16.8.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "cjs": "lib/dist/core.cjs.js",
  "meta": {
    "sizeLimit": "55 KB"
  },
  "sideEffects": false
}
