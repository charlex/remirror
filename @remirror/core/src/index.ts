export * from './extension';
export * from './extension-helpers';
export * from './extension-manager';
export * from './extension-types';
export * from './mark-extension';
export * from './node-extension';
export * from './nodes';

export * from '@remirror/core-constants';
export * from '@remirror/core-types';
export * from '@remirror/core-utils';
export * from '@remirror/core-helpers';
