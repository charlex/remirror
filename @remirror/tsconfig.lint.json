{
  "extends": "../support/tsconfig.base.json",
  "compilerOptions": {
    "declaration": false,
    "allowJs": true,
    "skipLibCheck": true,
    "types": [
      "jest",
      "jest-extended",
      "jest-axe",
      "@testing-library/jest-dom/extend-expect",
      "jest-emotion",
      "jest-enzyme",
      "jest-environment-enzyme"
    ]
  },
  "include": ["**/*.ts", "**/*.tsx", "**/*.js"],
  "exclude": ["lib", "**/lib", "node_modules/**"]
}
