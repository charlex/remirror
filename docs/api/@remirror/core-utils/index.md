<!-- Do not edit this file. It is automatically generated by API Documenter. -->

[Home](./index.md)

## API Reference

## Packages

|  Package | Description |
|  --- | --- |
|  [@remirror/core-utils](./core-utils.md) |  |

