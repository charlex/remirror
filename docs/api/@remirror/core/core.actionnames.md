<!-- Do not edit this file. It is automatically generated by API Documenter. -->

[Home](./index.md) &gt; [@remirror/core](./core.md) &gt; [ActionNames](./core.actionnames.md)

## ActionNames type

Utility type for pulling all the action names from a list

<b>Signature:</b>

```typescript
export declare type ActionNames<GExtension extends AnyExtension> = StringKey<ActionsFromExtensions<GExtension>>;
```
