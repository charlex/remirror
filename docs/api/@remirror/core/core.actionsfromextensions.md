<!-- Do not edit this file. It is automatically generated by API Documenter. -->

[Home](./index.md) &gt; [@remirror/core](./core.md) &gt; [ActionsFromExtensions](./core.actionsfromextensions.md)

## ActionsFromExtensions type

Utility type which receives an extension and provides the type of actions it makes available.

<b>Signature:</b>

```typescript
export declare type ActionsFromExtensions<GExtension extends AnyExtension> = UnionToIntersection<MapCommandToAction<GExtension['_C']>>;
```
