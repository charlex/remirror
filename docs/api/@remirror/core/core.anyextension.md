<!-- Do not edit this file. It is automatically generated by API Documenter. -->

[Home](./index.md) &gt; [@remirror/core](./core.md) &gt; [AnyExtension](./core.anyextension.md)

## AnyExtension type

Provides a type annotation which is applicable to any extension type.

<b>Signature:</b>

```typescript
export declare type AnyExtension = Extension<any, any>;
```
