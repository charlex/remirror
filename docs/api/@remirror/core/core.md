<!-- Do not edit this file. It is automatically generated by API Documenter. -->

[Home](./index.md) &gt; [@remirror/core](./core.md)

## core package

## Classes

|  Class | Description |
|  --- | --- |
|  [DocExtension](./core.docextension.md) | The parent node in a Prosemirror Schema. A representation of the doc is required in all editors  |
|  [Extension](./core.extension.md) | Extensions are fundamental to the way that Remirror works and they handle the management of similar concerns. |
|  [ExtensionManager](./core.extensionmanager.md) | A class to manage the extensions and prosemirror interactions of our editor. |
|  [MarkExtension](./core.markextension.md) | All mark extensions should be created from here. |
|  [NodeExtension](./core.nodeextension.md) | Defines an abstract class for creating extensions which can place nodes into the prosemirror state. |
|  [TextExtension](./core.textextension.md) | The default text passed into the prosemirror schema. |

## Interfaces

|  Interface | Description |
|  --- | --- |
|  [DocExtensionOptions](./core.docextensionoptions.md) |  |
|  [Extension](./core.extension.md) |  |
|  [ExtensionConstructor](./core.extensionconstructor.md) | An extension constructor |
|  [ExtensionListParams](./core.extensionlistparams.md) |  |
|  [ExtensionManagerData](./core.extensionmanagerdata.md) |  |
|  [ExtensionParams](./core.extensionparams.md) |  |
|  [ManagerParams](./core.managerparams.md) |  |
|  [OnTransactionManagerParams](./core.ontransactionmanagerparams.md) |  |
|  [PrioritizedExtension](./core.prioritizedextension.md) | Provides a priority value to the extension which determines the priority. |

## Variables

|  Variable | Description |
|  --- | --- |
|  [isExtension](./core.isextension.md) | Determines if the passed in extension is a any type of extension. |
|  [isExtensionManager](./core.isextensionmanager.md) | Checks to see whether this is an extension manager |
|  [isMarkExtension](./core.ismarkextension.md) | Determines if the passed in extension is a mark extension. Useful as a type guard where a particular type of extension is needed. |
|  [isNodeExtension](./core.isnodeextension.md) | Determines if the passed in extension is a node extension. Useful as a type guard where a particular type of extension is needed. |
|  [isPlainExtension](./core.isplainextension.md) | Checks whether the this is an extension and if it is a plain one |

## Type Aliases

|  Type Alias | Description |
|  --- | --- |
|  [ActionNames](./core.actionnames.md) | Utility type for pulling all the action names from a list |
|  [ActionsFromExtensions](./core.actionsfromextensions.md) | Utility type which receives an extension and provides the type of actions it makes available. |
|  [AnyExtension](./core.anyextension.md) | Provides a type annotation which is applicable to any extension type. |
|  [AnyExtensionManager](./core.anyextensionmanager.md) | The utility for capturing all the possible <code>ExtensionManager</code> type variations. |
|  [CommandsOfExtension](./core.commandsofextension.md) | Utility type for retrieving the commands provided by an extension. |
|  [DataOfExtension](./core.dataofextension.md) | Utility type for retrieving the extension data object made available from the extension. |
|  [ExtensionsFromManager](./core.extensionsfrommanager.md) | Retrieve the extensions from an <code>ExtensionManager</code>. |
|  [FlexibleExtension](./core.flexibleextension.md) | Either a PrioritizedExtension or the actual Extension. |
|  [FlexibleExtensions](./core.flexibleextensions.md) |  |
|  [FromFlexibleExtension](./core.fromflexibleextension.md) |  |
|  [HelperNames](./core.helpernames.md) | Utility type for pulling all the action names from a list |
|  [HelpersFromExtensions](./core.helpersfromextensions.md) | Utility type which receives an extension and provides the type of helpers it makes available. |
|  [HelpersOfExtension](./core.helpersofextension.md) | Utility type for retrieving the helpers provided by an extension. |
|  [InferFlexibleExtension](./core.inferflexibleextension.md) | Get the extension from a PrioritizedExtension. |
|  [InferFlexibleExtensionList](./core.inferflexibleextensionlist.md) | Get the extension types from a list of PrioritizedExtensions. |
|  [MapCommandToAction](./core.mapcommandtoaction.md) | A utility type which maps the passed in extension command in an action that is called via <code>manager.data.actions.commandName()</code>. |
|  [MapHelpers](./core.maphelpers.md) | A utility type which maps the passed in extension helpers to a method called with <code>manager.data.helpers.helperName()</code>. |
|  [MarkNames](./core.marknames.md) | A utility type for retrieving the name of an extension only when it's a mark extension. |
|  [NameOfExtension](./core.nameofextension.md) | Utility type for retrieving the name of an extension. |
|  [NodeNames](./core.nodenames.md) | A utility type for retrieving the name of an extension only when it's a node extension. |
|  [OptionsOfExtension](./core.optionsofextension.md) | Utility type for retrieving the extension options from an extension. |
|  [PlainNames](./core.plainnames.md) | A utility type for retrieving the name of an extension only when it's a plain extension. |
|  [ProsemirrorTypeOfExtension](./core.prosemirrortypeofextension.md) | Utility type for retrieving the prosemirror type of the extension which can be:<!-- -->- <code>NodeType</code> - <code>MarkType</code> - <code>never</code> |
|  [SchemaFromExtensions](./core.schemafromextensions.md) | Gets the schema from a list of extensions |
|  [TypeOfExtensionClass](./core.typeofextensionclass.md) | Retrieve the instance type from an ExtensionClass. |
|  [TypeOfExtensionClassList](./core.typeofextensionclasslist.md) | This is a utility type which allows for retrieving the instance types from an array of Extensions. |

