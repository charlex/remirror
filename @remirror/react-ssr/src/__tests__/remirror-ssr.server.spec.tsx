/**
 * @jest-environment node
 */

import React from 'react';

import { createTestManager, helpers, initialJson } from '@test-fixtures/schema-helpers';
import { renderToString } from 'react-dom/server';

import { ExtensionManager } from '@remirror/core';
import { RemirrorSSR } from '..';

let manager: ExtensionManager;

beforeEach(() => {
  manager = createTestManager();
});

test('should render the ssr component', () => {
  manager.init({ ...helpers, getState: () => state });
  const state = manager.createState({ content: initialJson });

  const htmlString = renderToString(
    <RemirrorSSR
      editable={true}
      attributes={{
        class: 'remirror',
      }}
      manager={manager}
      state={state}
    />,
  );
  expect(htmlString).toInclude('Better docs to come soon...');
  expect(htmlString).toMatchSnapshot();
});
