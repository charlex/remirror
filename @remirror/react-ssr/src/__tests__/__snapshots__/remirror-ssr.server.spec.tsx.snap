// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the ssr component 1`] = `
<div class="remirror"
     contenteditable="true"
     data-reactroot
>
  <p>
    Better docs to come soon...
  </p>
</div>
`;
